r/R/filesystem.R

```diff
@@ -232,6 +243,17 @@ LocalFileSystem$create <- function() {
 #' @export
 SubTreeFileSystem <- R6Class("SubTreeFileSystem", inherit = FileSystem)
 SubTreeFileSystem$create <- function(base_path, base_fs) {
-  xp <- fs___SubTreeFileSystem__create(base_path, base_fs)
+  xp <- fs___SubTreeFileSystem__create(fs_relative_path(base_path), base_fs)
   shared_ptr(SubTreeFileSystem, xp)
 }
+
+clean_path <- function(path) {
```

Comment:
This looks like things potentially better fixed with `fs::` cc @jimhester. I know `fs` has been downgraded to `Suggests` here, but I'm just worried we'd be missing cases with the implementation below.

Reply:
I didn't see the `fs` function that does the winslash normalization we need here, but more than happy to use it if it exists.